### DISCUSSION TITLE
How can we enable group:monorepo preset with automerge for npm only?

### DISCUSSION THREAD

**Discussion by moimael** [renovatebot/renovate#30238] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github, 37.433.0

### Please tell us more about your question or problem

Hi!

We use Renovate to update our NPM packages among others, and we use

`extends: ["group:monorepos"]`

to pick up renovate's [builtin list](https://docs.renovatebot.com/presets-group/) of known monorepo groups.

We want to activate automerge only for npm dependencies (we have other managers as well)

If I'd defined the groups myself, I could do something like

{
    groupName: "npm monorepo",
    matchPackagePatterns: [... however this is defined ...],
    matchUpdateTypes: ["minor", "patch"],
    automerge: true
},

But given that Renovate has already defined the group, I can't find any way in the documentation to alter the existing group.

Is this possible? or do I have to opt out of group:monorepos and declare all groups manually?

The grouping works if we create the PR manually through the dashboard, but it doesn't work for automerge.

Our config:

```
  extends: ["mergeConfidence:all-badges", "group:monorepos"],
  onboarding: false, // as we self-host, bypass default onboarding
  requireConfig: "ignored", // as we self-host, bypass finding config file as it's defined explicitely
  dependencyDashboard: true,
  dependencyDashboardApproval: true, // By default, no automated PR unless we explicitely ask for it
  repositories: ["<repo_name>"], // Necessary when running self-hosted Renovate
  gitAuthor: "some_author", // Same as user behind RENOVATE_TOKEN
  labels: ["dependencies"],
  prConcurrentLimit: 100,
  rangeStrategy: "update-lockfile",
  branchConcurrentLimit: 5,
  postUpdateOptions: ["yarnDedupeHighest"],

  // Ignore ios/android dependencies as they are handled by react-native
  gradle: { enabled: false },
  npm: {
    minimumReleaseAge: "3 days",
  },
  packageRules: [
    {
      matchManagers: ["npm"],
      dependencyDashboardApproval: false,
      extends: [":automergeMinor", ":automergeRequireAllStatusChecks", ":automergePr", ":prImmediately"],
      automergeSchedule: ["before 1pm every weekday except on friday"], // Time is in UTC 1pm = 3pm in Paris
    },
  ],
  prBodyTemplate: PR_BODY_TEMPLATE,
  prBodyColumns: COMMON_PR_BODY_COLUMNS,
  prBodyDefinitions: COMMON_PR_BODY_DEFINITIONS,

  // overwritten conf for vulnerability alerts
  vulnerabilityAlerts: {
    dependencyDashboardApproval: false, // Automated PRs for all vulnerability alerts
    addLabels: ["security"],
    prBodyDefinitions: {
      ...COMMON_PR_BODY_DEFINITIONS,
      Change:
        // add an easily searchable comment for each package to easily map the PR to a potential vulnerability alert
        `${COMMON_PR_BODY_DEFINITIONS["Change"]}<!-- dependency_vulnerability:{{{manager}}}:{{{depName}}}:{{{currentVersion}}}:{{{newVersion}}} -->`,
    },
  },
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
You only want to merge if they're (a) monorepos, and (b) npm? Why wouldn't you also merge non-monorepo npm updates?

**Reply by moimael** [reply]
I'd also like to merge non-monorepo npm updates. Currently that's all it does. It completely ignores monorepo updates (unless I trigger the update manually through the dependencies dashboard)

**Reply by rarkins** [reply]
You can write a package rule which says matchManagers=npm, matchUpdateTypes=patch,minor, and automerge=true

Any group will be automerged so long as all updates within that group are configured for automerge (e.g. all are non-major npm updates)

**Reply by moimael** [reply]
That's my current config, and it does work for non monorepo update. 

```
  packageRules: [
    {
      matchManagers: ["npm"],
      dependencyDashboardApproval: false,
      extends: [":automergeMinor", ":automergeRequireAllStatusChecks", ":automergePr", ":prImmediately"],
      automergeSchedule: ["before 1pm every weekday except on friday"], // Time is in UTC 1pm = 3pm in Paris
    },
  ],
```
What I don't get is how to get monorepo updates to automerge?

Globally I have

`  extends: ["mergeConfidence:all-badges", "group:monorepos"],`

But since it creates packageRules behind the scene, I cannot extend those group to have

`dependencyDashboardApproval: false,`

And trigger the automerge for monorepos deps. 

And I can't create a custom `packageRule` that extends "group:monorepos", I get a config error in this case saying it's not allowed.

**Reply by rarkins** [reply]
No that's not your current config, your config is different to what I described

**Reply by moimael** [reply]
I'm a bit confused, isn't it what `extends [":automergeMinor"]` does?

Which translates to:
```
{
  "lockFileMaintenance": {
    "automerge": true
  },
  "minor": {
    "automerge": true
  },
  "patch": {
    "automerge": true
  },
  "pin": {
    "automerge": true
  }
}
```

**Reply by moimael** [reply]
I guess the difference is `matchUpdateTypes` I'll try with this, thanks!

**Reply by rarkins** [reply, marked as answer]
There's multiple differences and they are all important
